File: pset6/cash/cash.py
from cs50 import get_float
import math

while True:
    change = get_float("Change owed: ")
    if change > 0:
        break

num25 = int(change / 0.25)
num10 = int(round((change % 0.25), 2) / 0.1)
num5 = int(round(round((change % 0.25), 2) % 0.1, 2) / 0.05)
num1 = int(round(round(round((change % 0.25), 2) % 0.1, 2) % 0.05, 2) / 0.01)

print(num25 + num10 + num5 + num1)

File: Course-1/problem_set_1 (C)/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

int main(void)
{
    float change;
    int convertedChange;
    int coinType[4] = { 25, 10, 5, 1 };

    int coinCounter = 0;

    // Get user input and validate it
    change = get_float("Change owed: ");
    while (change < 0)
    {
        change = get_float("Change owed: ");
    }

    convertedChange = (int)round(change * 100);

    for (int i = 0; i < sizeof coinType; i++)
    {
        coinCounter = coinCounter + floor(convertedChange / coinType[i]);
        convertedChange = convertedChange % coinType[i];
        if (convertedChange < 1)
        {
            break;
        }
    }

    printf("%i\n", coinCounter);

}